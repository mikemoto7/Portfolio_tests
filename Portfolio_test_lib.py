
from test_flags import _product_called_from_test
_product_called_from_test = True


import sys
sys.path.append("../lib_for_test")

from general_test_lib import do_test_runstring
import general_test_lib

general_test_lib.product_dir_path = "../Portfolio"







