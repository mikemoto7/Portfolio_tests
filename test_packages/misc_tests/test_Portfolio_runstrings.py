import pytest

from Portfolio_test_lib import do_test_runstring


# @pytest.mark.usefixtures("oneTimeSetUp", "setUp")
@pytest.mark.usefixtures()
class TestPortfolio():

    # @pytest.fixture(autouse=True)
    # def classSetup(self, setup):
    #     pass



    # Testcases interactive:

    '''
    def test_runstring_cl(self):
        do_test_runstring("Portfolio.py --cl")

    def test_runstring_cl_l(self):
        do_test_runstring("Portfolio.py --cl l")

    def test_runstring_cl_7(self):
        do_test_runstring("Portfolio.py --cl 7")

    def test_runstring_cl_debug_1(self):
        do_test_runstring("Portfolio.py --cl 7 --debug 1")
    '''


    # Testcases from docstring:


    def test_runstring_price_web_update(self):
        do_test_runstring("Portfolio.py --price_web_update --pfile demo_data_file.xml")

    def test_runstring_ac_web_update(self):
        do_test_runstring("Portfolio.py --ac_web_update --pfile demo_data_file.xml")

    def test_runstring_stats(self):
        do_test_runstring("Portfolio.py --stats --pfile demo_data_file.xml")

    def test_runstring_xml(self):
        do_test_runstring("Portfolio.py --xml --pfile demo_data_file.xml")

    def test_runstring_price_web_update_xml(self):
        do_test_runstring("Portfolio.py --price_web_update --xml --pfile demo_data_file.xml")

    def test_runstring_pt_show(self):
        do_test_runstring("Portfolio.py --pt show --pfile demo_data_file.xml")

    def test_runstring_pt_show_percent(self):
        do_test_runstring("Portfolio.py --pt show_percent --pfile demo_data_file.xml")

    def test_runstring_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --pt show_ac_misc_totals --pfile demo_data_file.xml")

    def test_runstring_pt_lp(self):
        do_test_runstring("Portfolio.py --pt lp --pfile demo_data_file.xml")

    def test_runstring_pt_add(self):
        do_test_runstring("Portfolio.py --pt add:'Small Value',Non_Retirement,50000:'Small Value',Retirement,20000:Healthcare,Retirement,50000 --pfile demo_data_file.xml")

    def test_runstring_inv_show(self):
        do_test_runstring("Portfolio.py --inv show --pfile demo_data_file.xml")

    def test_runstring_inv_diff_1file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml")

    def test_runstring_inv_diff_2file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml1,demo_data_file.xml2")

    def test_runstring_inv_lots(self):
        do_test_runstring("Portfolio.py --inv lots --pfile demo_data_file.xml")

    def test_runstring_all_not_silent(self):
        do_test_runstring("Portfolio.py --all --pfile demo_data_file.xml")

    def test_runstring_all_silent(self):
        do_test_runstring("Portfolio.py --all --silent --pfile demo_data_file.xml")

    def test_runstring_noindent(self):
        do_test_runstring("Portfolio.py --noindent  --all  --pfile demo_data_file.xml")

    def test_runstring_h(self):
        do_test_runstring("Portfolio.py --h")

    def test_runstring_xmlfile(self):
        do_test_runstring("Portfolio.py --xmlfile  --pfile demo_data_file.xml")

    def test_runstring_pdt(self):
        do_test_runstring("Portfolio.py --pdt --pfile demo_data_file.xml")

    def test_runstring_debug_1(self):
        do_test_runstring("Portfolio.py --debug 1  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_2(self):
        do_test_runstring("Portfolio.py --debug 2  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_12(self):
        do_test_runstring("Portfolio.py --debug 12  --all  --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_show(self):
        do_test_runstring("Portfolio.py --price_web_update --inv show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_ac_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --ac_web_update --pt show%lp%add:Small_Value,Retirement,13000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_lp(self):
        do_test_runstring("Portfolio.py --price_web_update --pt lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Large_Blend,Retirement,13000%show_percent%lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_diff(self):
        do_test_runstring("Portfolio.py --price_web_update --inv diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent_bigger_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_ac_misc_totals --pfile demo_data_file.xml")






