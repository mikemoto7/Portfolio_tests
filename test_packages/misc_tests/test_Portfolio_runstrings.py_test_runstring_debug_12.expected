
Processing: user_runstring = ./Portfolio.py --debug 12 --all --pfile demo_data_file.xml

Expanded to: --debug 12 --price_web_update  --ac_web_update  --pt show_percent --pt show_ac_misc_totals --inv lots --inv diff --pfile demo_data_file.xml


Processing: --pt show_percent
-----------------------------------------------------------------------------------------------------------
                          Tot        Tot    Tot        Ret        Ret    Ret    Non_Ret    Non_Ret  Non_Ret
Asset_Class         inv_count        amt    pct  inv_count        amt    pct  inv_count        amt      pct
Energy                      4   97837.40  11.39          3   77939.40  15.04          1   19898.00     5.84
Global                      6   57632.60   6.71          4   48694.30   9.40          2    8938.30     2.62
Healthcare                  2   25640.10   2.98          1    4135.50   0.80          1   21504.60     6.31
Individual_Stocks           1   25878.60   3.01          0       0.00   0.00          1   25878.60     7.59
Large_Blend                 3   43508.60   5.06          2   23145.60   4.47          1   20363.00     5.97
Large_Growth                4  107245.50  12.48          2   77404.00  14.94          2   29841.50     8.75
Large_Value                 3   47526.30   5.53          2   32897.70   6.35          1   14628.60     4.29
Mid_Blend_Aggress           3   73973.90   8.61          0       0.00   0.00          3   73973.90    21.70
Mid_Growth                  5   85482.90   9.95          3   69038.00  13.32          2   16444.90     4.82
Mid_Value                   5   54082.80   6.30          3   30378.20   5.86          2   23704.60     6.95
REIT                        5  114252.00  13.30          4   79805.80  15.40          1   34446.20    10.10
Small_Blend                 2   23496.80   2.74          1    4477.20   0.86          1   19019.60     5.58
Small_Growth                4   82504.50   9.60          3   67261.80  12.98          1   15242.70     4.47
Small_Value                 2   20006.80   2.33          1    2970.00   0.57          1   17036.80     5.00
Totals                     49  859068.80  99.99         29  518147.50  99.99         20  340921.30    99.99
                                                                                                           

Processing: --pt show_ac_misc_totals
--------------------------------------------------------------------------------------------------
                     Tot        Tot     Tot    Ret        Ret     Ret  Non_Ret    Non_Ret  Non_Ret
                   count        amt     pct  count        amt     pct    count        amt      pct
Asset Classes         14                        12                          14                    
                                                                                                  
                                                                                                  
Vanguard              42  696708.30   81.10     25  401922.60   77.57       17  294785.70    86.47
Fidelity               6  136481.90   15.89      4  116224.90   22.43        2   20257.00     5.94
Schwab                 1   25878.60    3.01      0       0.00    0.00        1   25878.60     7.59
Totals                49  859068.80  100.00     29  518147.50  100.00       20  340921.30   100.00
                                                                                                  
retirement_class      49  859068.80  100.00     29  518147.50   60.32       20  340921.30    39.68
                                                                                                  
owner2                26  435326.20   50.67     16  255691.50   49.35       10  179634.70    52.69
owner1                23  423742.60   49.33     13  262456.00   50.65       10  161286.60    47.31
Totals                49  859068.80  100.00     29  518147.50  100.00       20  340921.30   100.00
                                                                                                  


Processing: --inv lots
-----------------------------------------------------------------------------------------------------------------------------------------
Asset                           Symb  Own  iID  Account_ID  Ret_Class   Curr_Tot   CB_Date  CB_Qty  CB_ShPr     CB_Tot   GainLoss    GL_%
---Energy---                                                                                                                             
Vanguard Energy                VGELX  os1   20       15777    Non_Ret   19898.00  00000000   160.0     0.00       0.00   19898.00    0.00
Vanguard Energy                VGENX  os1   42       35777   Trad_IRA   21208.00  00000000   380.0     0.00       0.00   21208.00    0.00
Vanguard Energy                VGENX  os1   52       45777   Roth_IRA   26510.00  00000000   490.0     0.00       0.00   26510.00    0.00
Vanguard Energy                VGENX  os2   59       52777   Roth_IRA   30221.40  00000000   560.0     0.00       0.00   30221.40    0.00
---Global---                                                                                                                             
Fidelity Canada                FICDX  os2   39       34777   Trad_IRA   19995.30  00000000   370.0    47.06   17412.20    2583.10   14.84
Vanguard Global Equity         VHGEX  os2    5        4777    SEP_IRA    2348.10  00000000    50.0     0.00       0.00    2348.10    0.00
Vanguard Global Equity         VHGEX  os2   23       18777    Non_Ret    6000.70  00000000   190.0     0.00   63359.37  -57358.33  -90.53
Vanguard Global Equity         VHGEX  os2   47       40777   Trad_IRA   11740.50  00000000   430.0     0.00       0.00   11740.50    0.00
Vanguard Global Equity         VHGEX  os1   58       51777   Trad_IRA   14610.40  00000000   550.0     0.00       0.00   14610.40    0.00
Vanguard Global Min Volatile   VMVFX  os1   24       19777    Non_Ret    2937.60  00000000   200.0     0.00       0.00    2937.60    0.00
---Healthcare---                                                                                                                         
Vanguard Healthcare            VGHAX  os2    1         777    SEP_IRA    4135.50  00000000    10.0     0.00       0.00    4135.50    0.00
Vanguard Healthcare            VGHAX  os1   26       21777    Non_Ret   21504.60  00000000   220.0     0.00       0.00   21504.60    0.00
---Individual_Stocks---                                                                                                                  
Netflix                         NFLX  os1   18       13777    Non_Ret   25878.60  00000000   140.0     0.00       0.00   25878.60    0.00
---Large_Blend---                                                                                                                        
Vanguard Dividend Growth       VDIGX  os2   45       38777   Trad_IRA   10367.30  00000000   410.0    17.72    7265.20    3102.10   42.70
Vanguard Dividend Growth       VDIGX  os2   55       48777   Trad_IRA   12778.30  00000000   520.0     0.00       0.00   12778.30    0.00
Vanguard Total Stock Mkt Idx   VTSAX  os2   35       30777    Non_Ret   20363.00  00000000   320.0     0.00   59254.97  -38891.03  -65.63
---Large_Growth---                                                                                                                       
Fidelity Contrafund            FCNTX  os1   13        9777    Non_Ret   14644.00  00000000   100.0    57.91    5791.00    8853.00  152.88
Fidelity Contrafund            FCNTX  os1   36       31777   Trad_IRA   37656.00  00000000   330.0    19.75    6517.50   31138.50  477.77
Fidelity Contrafund            FCNTX  os1   38       33777  Trad_401K   39748.00  00000000   350.0     0.00       0.00   39748.00    0.00
Vanguard Growth Idx            VIGAX  os2   25       20777    Non_Ret   15197.50  00000000   210.0    54.40   11424.00    3773.50   33.03
---Large_Value---                                                                                                                        
Vanguard Equity Income         VEIRX  os2   21       16777    Non_Ret   14628.60  00000000   170.0     0.00   67872.31  -53243.29  -78.45
Vanguard Equity Income         VEIPX  os1   46       39777   Trad_IRA   14621.20  00000000   420.0     0.00       0.00   14621.20    0.00
Vanguard Equity Income         VEIPX  os2   57       50777   Trad_IRA   18276.50  00000000   540.0    27.06   14612.40    3664.10   25.08
---Mid_Blend_Aggress---                                                                                                                  
Vanguard Extended Mkt Idx      VEXAX  os1   22       17777    Non_Ret   16662.80  00000000   180.0     0.00       0.00   16662.80    0.00
Vanguard Mid-Cap Idx           VIMAX  os2   27       22777    Non_Ret   46213.20  00000000   230.0     0.00   34955.87   11257.33   32.20
Vanguard Strategic Equity      VSEQX  os2   33       28777    Non_Ret   11097.90  00000000   300.0     0.00   33515.16  -22417.74  -66.89
---Mid_Growth---                                                                                                                         
Fidelity New Millennium        FMILX  os2   14       10777    Non_Ret    5613.00  00000000   110.0    28.76    3163.60    2449.40   77.42
Vanguard Capital Opportunity   VHCOX  os2   19       14777    Non_Ret   10831.90  00000000   150.0    41.11    6166.50    4665.40   75.66
Vanguard Capital Opportunity   VHCAX  os1   50       43777   Rollover   63168.00  00000000   460.0     0.00       0.00   63168.00    0.00
Vanguard Mid-Cap Growth        VMGRX  os1    8        7777   Trad_IRA    2817.60  00000000    80.0     0.00       0.00    2817.60    0.00
Vanguard Mid-Cap Growth        VMGRX  os2    9        8777   Trad_IRA    3052.40  00000000    90.0     0.00       0.00    3052.40    0.00
---Mid_Value---                                                                                                                          
Fidelity Low-Priced Stock      FLPSX  os2   37       32777   Trad_IRA   18825.60  00000000   340.0    12.68    4311.20   14514.40  336.67
Vanguard Capital Value         VCVLX  os1   44       37777   Trad_IRA    5161.80  00000000   400.0     0.00       0.00    5161.80    0.00
Vanguard Capital Value         VCVLX  os1   54       47777   Trad_IRA    6390.80  00000000   510.0     0.00       0.00    6390.80    0.00
Vanguard Mid-Cap Value         VMVAX  os1   28       23777    Non_Ret   14677.60  00000000   240.0     0.00       0.00   14677.60    0.00
Vanguard Selected Value        VASVX  os1   30       25777    Non_Ret    9027.00  00000000   260.0     0.00       0.00    9027.00    0.00
---REIT---                                                                                                                               
Vanguard REIT                  VGSLX  os1    4        3777    SEP_IRA    9502.40  00000000    40.0     0.00       0.00    9502.40    0.00
Vanguard REIT                  VGSIX  os2    7        6777   Trad_IRA    3062.40  00000000    70.0     0.00       0.00    3062.40    0.00
Vanguard REIT                  VGSLX  os2   29       24777    Non_Ret   34446.20  00000000   250.0     0.00   72725.62  -38279.58  -52.64
Vanguard REIT                  VGSIX  os2   43       36777   Trad_IRA   11414.40  00000000   390.0     0.00       0.00   11414.40    0.00
Vanguard REIT                  VGSLX  os2   49       42777   Trad_IRA   55826.60  00000000   450.0     0.00       0.00   55826.60    0.00
---Small_Blend---                                                                                                                        
Vanguard Small Idx             NAESX  os2    3        2777   Trad_IRA    4477.20  00000000    30.0    53.73    1611.90    2865.30  177.76
Vanguard Tax-Managed Sml Val   VTMSX  os1   34       29777    Non_Ret   19019.60  00000000   310.0     0.00       0.00   19019.60    0.00
---Small_Growth---                                                                                                                       
Vanguard Small Growth          VSGAX  os2   51       44777   Rollover   24093.30  00000000   470.0     0.00   10080.00   14013.30  139.02
Vanguard Small Growth          VSGAX  os2   53       46777   Roth_IRA   25076.70  00000000   500.0     0.00       0.00   25076.70    0.00
Vanguard Small Growth Idx      VSGAX  os2   31       26777    Non_Ret   15242.70  20110921   270.0    19.73    5524.40    9718.30  175.92
Vanguard Small Growth Idx      VISGX  os1   48       41777   Trad_IRA   18091.80  00000000   440.0    21.47    9446.80    8645.00   91.51
---Small_Value---                                                                                                                        
Vanguard Small-Cap Val Idx     VISVX  os1    6        5777   Trad_IRA    2970.00  00000000    60.0     0.00       0.00    2970.00    0.00
Vanguard Small-Cap Val Idx     VSIAX  os1   32       27777    Non_Ret   17036.80  00000000   290.0     0.00       0.00   17036.80    0.00
---Totals---                                                                                                                             
49                                                                     859068.80                             435010.00  424058.80        

Processing: --price_web_update

Processing: --ac_web_update


Processing: --inv diff
------------------------------------------------------------------------------------------------------------------------
Asset                           Symb  Own  iID  Account_ID  Ret_Class  Curr_Tot1  Curr_Tot2   ShPr1   ShPr2  Diff  Diff%
---Energy---                                                                                                            
Vanguard Energy                VGELX  os1   20       15777    Non_Ret   19898.00   19898.00   99.49   99.49  0.00    0.0
Vanguard Energy                VGENX  os1   42       35777   Trad_IRA   21208.00   21208.00   53.02   53.02  0.00    0.0
Vanguard Energy                VGENX  os1   52       45777   Roth_IRA   26510.00   26510.00   53.02   53.02  0.00    0.0
Vanguard Energy                VGENX  os2   59       52777   Roth_IRA   30221.40   30221.40   53.02   53.02  0.00    0.0
---Global---                                                                                                            
Fidelity Canada                FICDX  os2   39       34777   Trad_IRA   19995.30   19995.30   51.27   51.27  0.00    0.0
Vanguard Global Equity         VHGEX  os2    5        4777    SEP_IRA    2348.10    2348.10   26.09   26.09  0.00    0.0
Vanguard Global Equity         VHGEX  os2   23       18777    Non_Ret    6000.70    6000.70   26.09   26.09  0.00    0.0
Vanguard Global Equity         VHGEX  os2   47       40777   Trad_IRA   11740.50   11740.50   26.09   26.09  0.00    0.0
Vanguard Global Equity         VHGEX  os1   58       51777   Trad_IRA   14610.40   14610.40   26.09   26.09  0.00    0.0
Vanguard Global Min Volatile   VMVFX  os1   24       19777    Non_Ret    2937.60    2937.60   12.24   12.24  0.00    0.0
---Healthcare---                                                                                                        
Vanguard Healthcare            VGHAX  os2    1         777    SEP_IRA    4135.50    4135.50   82.71   82.71  0.00    0.0
Vanguard Healthcare            VGHAX  os1   26       21777    Non_Ret   21504.60   21504.60   82.71   82.71  0.00    0.0
---Individual_Stocks---                                                                                                 
Netflix                         NFLX  os1   18       13777    Non_Ret   25878.60   25878.60  143.77  143.77  0.00    0.0
---Large_Blend---                                                                                                       
Vanguard Dividend Growth       VDIGX  os2   45       38777   Trad_IRA   10367.30   10367.30   24.11   24.11  0.00    0.0
Vanguard Dividend Growth       VDIGX  os2   55       48777   Trad_IRA   12778.30   12778.30   24.11   24.11  0.00    0.0
Vanguard Total Stock Mkt Idx   VTSAX  os2   35       30777    Non_Ret   20363.00   20363.00   58.18   58.18  0.00    0.0
---Large_Growth---                                                                                                      
Fidelity Contrafund            FCNTX  os1   13        9777    Non_Ret   14644.00   14644.00  104.60  104.60  0.00    0.0
Fidelity Contrafund            FCNTX  os1   36       31777   Trad_IRA   37656.00   37656.00  104.60  104.60  0.00    0.0
Fidelity Contrafund            FCNTX  os1   38       33777  Trad_401K   39748.00   39748.00  104.60  104.60  0.00    0.0
Vanguard Growth Idx            VIGAX  os2   25       20777    Non_Ret   15197.50   15197.50   60.79   60.79  0.00    0.0
---Large_Value---                                                                                                       
Vanguard Equity Income         VEIRX  os2   21       16777    Non_Ret   14628.60   14628.60   69.66   69.66  0.00    0.0
Vanguard Equity Income         VEIPX  os1   46       39777   Trad_IRA   14621.20   14621.20   33.23   33.23  0.00    0.0
Vanguard Equity Income         VEIPX  os2   57       50777   Trad_IRA   18276.50   18276.50   33.23   33.23  0.00    0.0
---Mid_Blend_Aggress---                                                                                                 
Vanguard Extended Mkt Idx      VEXAX  os1   22       17777    Non_Ret   16662.80   16662.80   75.74   75.74  0.00    0.0
Vanguard Mid-Cap Idx           VIMAX  os2   27       22777    Non_Ret   46213.20   46213.20  171.16  171.16  0.00    0.0
Vanguard Strategic Equity      VSEQX  os2   33       28777    Non_Ret   11097.90   11097.90   33.63   33.63  0.00    0.0
---Mid_Growth---                                                                                                        
Fidelity New Millennium        FMILX  os2   14       10777    Non_Ret    5613.00    5613.00   37.42   37.42  0.00    0.0
Vanguard Capital Opportunity   VHCOX  os2   19       14777    Non_Ret   10831.90   10831.90   57.01   57.01  0.00    0.0
Vanguard Capital Opportunity   VHCAX  os1   50       43777   Rollover   63168.00   63168.00  131.60  131.60  0.00    0.0
Vanguard Mid-Cap Growth        VMGRX  os1    8        7777   Trad_IRA    2817.60    2817.60   23.48   23.48  0.00    0.0
Vanguard Mid-Cap Growth        VMGRX  os2    9        8777   Trad_IRA    3052.40    3052.40   23.48   23.48  0.00    0.0
---Mid_Value---                                                                                                         
Fidelity Low-Priced Stock      FLPSX  os2   37       32777   Trad_IRA   18825.60   18825.60   50.88   50.88  0.00    0.0
Vanguard Capital Value         VCVLX  os1   44       37777   Trad_IRA    5161.80    5161.80   12.29   12.29  0.00    0.0
Vanguard Capital Value         VCVLX  os1   54       47777   Trad_IRA    6390.80    6390.80   12.29   12.29  0.00    0.0
Vanguard Mid-Cap Value         VMVAX  os1   28       23777    Non_Ret   14677.60   14677.60   52.42   52.42  0.00    0.0
Vanguard Selected Value        VASVX  os1   30       25777    Non_Ret    9027.00    9027.00   30.09   30.09  0.00    0.0
---REIT---                                                                                                              
Vanguard REIT                  VGSLX  os1    4        3777    SEP_IRA    9502.40    9502.40  118.78  118.78  0.00    0.0
Vanguard REIT                  VGSIX  os2    7        6777   Trad_IRA    3062.40    3062.40   27.84   27.84  0.00    0.0
Vanguard REIT                  VGSLX  os2   29       24777    Non_Ret   34446.20   34446.20  118.78  118.78  0.00    0.0
Vanguard REIT                  VGSIX  os2   43       36777   Trad_IRA   11414.40   11414.40   27.84   27.84  0.00    0.0
Vanguard REIT                  VGSLX  os2   49       42777   Trad_IRA   55826.60   55826.60  118.78  118.78  0.00    0.0
---Small_Blend---                                                                                                       
Vanguard Small Idx             NAESX  os2    3        2777   Trad_IRA    4477.20    4477.20   63.96   63.96  0.00    0.0
Vanguard Tax-Managed Sml Val   VTMSX  os1   34       29777    Non_Ret   19019.60   19019.60   55.94   55.94  0.00    0.0
---Small_Growth---                                                                                                      
Vanguard Small Growth          VSGAX  os2   51       44777   Rollover   24093.30   24093.30   49.17   49.17  0.00    0.0
Vanguard Small Growth          VSGAX  os2   53       46777   Roth_IRA   25076.70   25076.70   49.17   49.17  0.00    0.0
Vanguard Small Growth Idx      VSGAX  os2   31       26777    Non_Ret   15242.70   15242.70   49.17   49.17  0.00    0.0
Vanguard Small Growth Idx      VISGX  os1   48       41777   Trad_IRA   18091.80   18091.80   39.33   39.33  0.00    0.0
---Small_Value---                                                                                                       
Vanguard Small-Cap Val Idx     VISVX  os1    6        5777   Trad_IRA    2970.00    2970.00   29.70   29.70  0.00    0.0
Vanguard Small-Cap Val Idx     VSIAX  os1   32       27777    Non_Ret   17036.80   17036.80   53.24   53.24  0.00    0.0
---Stats---                                                                                                             
Totals                            49                                   859068.80  859068.80                  0.00       
Totals diff percent                                                                                                  0.0
Avg diff percent                                                                                                    0.00
Count pos diff percent                                                                                                 0
Count neg diff percent                                                                                                 0
Count neutral diff percent                                                                                            49

